src/Chapter2/Exam.hs

```diff
@@ -0,0 +1,265 @@
+module Chapter2.Practice3 where
+import Data.List
+import Data.Char
+
+
+score :: [(String,[(String,Int)])]
+score=[ ("Mojo", [("English", 84), ("Chemisty", 80), ("Physics", 95), ("Geography", 75)])
+    , ("John Doe", [("Chemisty", 80), ("Physics", 95), ("Geography", 75)])
+    , ("Captain Jack", [("Chemisty", 66), ("Phsyics", 33), ("Geography", 56)]) -- Note the spelling error in "Physics"
+    , ("John Doe", [("Chemisty", 90), ("Economics", 45), ("Geography", 56)]) -- Note that "John Doe" is a repeat entry
+    , ("Bahubali", [("Hindi", 45), ("Biology", -90), ("Geography", -75)]) -- Note the negative marks in "Biology" & "Geography"
+    , ("Dexter", [("Tamil", 110), ("Biology", 100), ("Geography", 100)]) -- Note that marks in "Tamil" are greater than 100
+    ]
+
+validsubject :: [String]
+validsubject = ["English", "Geography", "Physics", "Chemistry", "Economics", "Computer Science"]
+
+type InputScore = [(String,[(String,Int)])]
+type Subject = String
+type SubjectAverage = (String,Float)     
+
+averageMarks :: InputScore ->ValidSubjects -> Subject -> SubjectAverage             --1st
+averageMarks score validsubject subject =let listSubMarks =  (filter (\x -> (fst x == subject)) (foldl' (\arr x -> arr ++ (snd x)) [] (allValidNames score validsubject) ) )
+                        in (subject, fromIntegral( (foldl'(\sum y -> (sum+snd y) ) 0 listSubMarks ) `div` length(listSubMarks)  ) )
+
+
+type ValidSubjects = [String]
+type StudentName = String
+type Marks = Int
+type Validate = (Bool,String)                                                                                                   --type Validate = Bool            --(Bool,String)                   --changed for 4
+
+
+validateMarksSubjects :: InputScore -> ValidSubjects -> StudentName -> Subject -> Validate
+validateMarksSubjects score validsubject studentName subjectName = let validMarks = snd (head (filter (\x -> (fst x == subjectName)) (snd (head (filter (\x -> (fst x) == studentName) score)))   ))
```

Comment:
this indentation is not good for readability. Put the let binding in the next line so that you code in indented in the following manner:

```
validateMarksSubjects score validsubject studentName subjectName =
  let validMarks = ...
  in ...
```